### PR TITLE
Fixes #7823: Added possibility to limit request by ajax in yii\filters\AccessRule

### LINKED ISSUE #7823: behavior access only ajax
hi, i have a question

how can block in the behavior for give access to the action ONLY by ajax ?

any idea or i need create a custom filter for this case

for example
public function behaviors()
    {
        return [  
            'ajax' => [
                'class' => AjaxFilter::className(),
                'only' => ['action1','action2'],  
            ]
        ];
    }
 and if you try enter with URL like (http://localhost/controller1/action1 you get 403

### PR DESCRIPTION
**Pull request by dmirogin**
Added possibility to limit request by ajax in yii\filters\AccessRule

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #7823

### CODE REVIEW

**Comment by cebe** [comment]
this is not really the job of AccessFilter but something like VerbFilter. Its not "Forbidden" but "Invalid Request".

**Comment by dmirogin** [comment]
Yes, really better make separate filter class.
@cebe Do you have other sight about utility this proposal ?

**Comment by cebe** [comment]
> @cebe Do you have other sight about utility this proposal ?

sorry, I do not understand that question, please rephrase :)

**Comment by dmirogin** [comment]
@cebe Guess i didn't understand you correctly from this sentence
> Except from that your code looks great to me, just not whats needed imo

I thought you want other changes except making separate filter class. Never mind so =).

**Comment by cebe** [comment]
Nope. just wanted to point out that I like your style of contribution :)

**Comment by SilverFire** [comment]
I agree with @cebe that it's not a responsibility of AccessRule class but I don't think we should add this feature to VerbFilter because it's not about HTTP request verb.

How about a separate AjaxFilter?

```php
[

**Comment by cebe** [comment]
yep, additional filter was also discussed in the Issue and is the best option imo.

File: framework/filters/AjaxFilter.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * @author Dmitry Dorogin <dmirogin@ya.ru>
+ * @since 2.0.13
+ */
+class AjaxFilter extends ActionFilter
+{
+    /**
+     * @var string the message to be displayed when request isn't ajax
+     */
+    public $errorMessage = 'Request must be XMLHttpRequest only.';
```
Comment (samdark) [inline framework/filters/AjaxFilter.php]: Request must be XMLHttpRequest.

**Review by samdark — APPROVED** [review]

**Review by dynasource — APPROVED** [review]

**Review by cebe — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**